src/main/java/com/example/ludogoriesoft/lukeriaerpapi/services/ProductService.java

```diff
@@ -65,9 +74,34 @@ private void validateProductDTO(ProductDTO productDTO) {
             throw new ValidationException("Package ID cannot be null");
         }
     }
+
     public void deleteProduct(Long id) throws ChangeSetPersister.NotFoundException {
         Product product = productRepository.findByIdAndDeletedFalse(id).orElseThrow(ChangeSetPersister.NotFoundException::new);
         product.setDeleted(true);
         productRepository.save(product);
     }
+
+    public ProductDTO produceProduct(Long productId, int producedQuantity) throws ChangeSetPersister.NotFoundException {
+        Optional<Product> product = productRepository.findByIdAndDeletedFalse(productId);
+        if (product.isPresent()) {
+            product.get().setAvailableQuantity(product.get().getAvailableQuantity() + producedQuantity);
+            productRepository.save(product.get());
+            Optional<Package> aPackage = packageRepository.findByIdAndDeletedFalse(product.get().getPackageId().getId());
+            if (aPackage.isPresent()) {
+                aPackage.get().setAvailableQuantity(aPackage.get().getAvailableQuantity() - producedQuantity);
+                packageRepository.save(aPackage.get());
+            }
+            Optional<Plate> plate = plateRepository.findByIdAndDeletedFalse(aPackage.get().getPlateId().getId());
+            if (plate.isPresent()) {
+                plate.get().setAvailableQuantity(plate.get().getAvailableQuantity() - producedQuantity);
+                plateRepository.save(plate.get());
+            }
+            Optional<Carton> carton = cartonRepository.findByIdAndDeletedFalse(aPackage.get().getCartonId().getId());
+            if (carton.isPresent()) {
+                carton.get().setAvailableQuantity(carton.get().getAvailableQuantity() - producedQuantity * aPackage.get().getPiecesCarton());
+                cartonRepository.save(carton.get());
```

Comment:
I think for a correct calculation we need to divide the _**producedQuantity
/ aPackage.get().getPiecesCarton()**_